akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
+        case (_, signal) ⇒
+          probe.ref ! GotChildSignal(signal)
+          Behavior.same
+      }).decorate
+      val actor = spawn(
+        (Behaviors.immutablePartial[Command] {
+          case (ctx, MakeChild) ⇒
+            val childRef = ctx.spawn(child, "A")
+            ctx.watch(childRef)
+            probe.ref ! ChildMade(childRef)
+            Behaviors.same
+          case (_, Inert) ⇒
+            probe.ref ! InertEvent
+            Behaviors.immutable[Command] {
+              case (_, _) ⇒ Behaviors.unhandled
+            } onSignal {
+              case (_, Terminated(_)) ⇒ Behaviors.unhandled
+              case (_, signal) ⇒
+                probe.ref ! GotSignal(signal)
+                Behaviors.same
+            }
+        } onSignal {
+          case (_, signal) ⇒
+            probe.ref ! GotSignal(signal)
+            Behaviors.same
+        }).decorate
+      )
+      actor ! MakeChild
+      val childRef = probe.expectMessageType[ChildMade].ref
+      actor ! Inert
+      probe.expectMessage(InertEvent)
+      childRef ! Stop
+      probe.expectMessage(GotChildSignal(PostStop))
+      probe.expectMessage(GotSignal(PostStop))
     }
 
     "return the right context info" in {
-      sync(setup("ctx20") { (ctx, startWith) ⇒
-        startWith.keep(_ ! GetInfo(ctx.self))
-          .expectMessage(expectTimeout) {
-            case (msg: Info, subj) ⇒
-              msg.self should ===(subj)
-              msg.system should ===(system)
-            case (other, _) ⇒
-              fail(s"$other was not an Info(...)")
-          }
-      })
+      type Info = (ActorSystem[Nothing], ActorRef[String])
+      val probe = TestProbe[Info]
+      val actor = spawn(Behaviors.immutablePartial[String] {
+        case (ctx, "info") ⇒
+          probe.ref ! (ctx.system → ctx.self)
+          Behaviors.same
+      }.decorate)
+      actor ! "info"
+      probe.expectMessage((system, actor))
     }
 
     "return right info about children" in {
-      sync(setup("ctx21") { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith
-          .mkChild(Some("B"), ctx.spawnMessageAdapter(ChildEvent), self)
-          .stimulate(_._1 ! GetChild("A", self), _ ⇒ Child(None))
-          .stimulate(_._1 ! GetChild("B", self), x ⇒ Child(Some(x._2)))
-          .stimulate(_._1 ! GetChildren(self), x ⇒ Children(Set(x._2)))
-      })
+      type Children = Seq[ActorRef[Nothing]]
+      val probe = TestProbe[Children]()
+      val actor = spawn(Behaviors.immutablePartial[String] {
+        case (ctx, "B") ⇒
+          ctx.spawn(Behaviors.empty, "B")
+          probe.ref ! ctx.child("B").toSeq
+          Behaviors.same
+        case (ctx, "all") ⇒
+          probe.ref ! ctx.children.toSeq
+          Behaviors.same
+        case (ctx, name) ⇒
+          probe.ref ! ctx.child(name).toSeq
+          Behaviors.same
+      }.decorate)
+      actor ! "B"
+      val children = probe.expectMessageType[Children]
+      actor ! "A"
```

Comment:
maybe call it "get" and explicitly match on it instead to not make it confusing with "B" spawning a child